### ISSUE TITLE
Support for additional detector_types in azurerm_monitor_smart_detector_alert_rule

### ISSUE THREAD

**Issue by gjermystreeva** [hashicorp/terraform-provider-azurerm#13997]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure Monitor App Insights now supports additional detector types "RequestPerformanceDegradationDetector", "DependencyPerformanceDegradationDetector", "ExceptionVolumeChangedDetector", "TraceSeverityDetector", "MemoryLeakDetector" as smart detection alerts.

See [Migrate Azure Monitor Application Insights smart detection to alerts (Preview)](https://docs.microsoft.com/en-gb/azure/azure-monitor/alerts/alerts-smart-detections-migration)

Feature request is to allow these additional detector_type s (above and beyond supported FailureAnomaliesDetector) on the azurerm_monitor_smart_detector_alert_rule resource.

### New or Affected Resource(s)

* azurerm_monitor_smart_detector_alert_rule

### Potential Terraform Configuration

For example the following should be possible to create
```hcl
resource "azurerm_monitor_smart_detector_alert_rule" "smart_detector_alert_rule" {

  for_each = toset(["RequestPerformanceDegradationDetector", "DependencyPerformanceDegradationDetector", "ExceptionVolumeChangedDetector", "TraceSeverityDetector", "MemoryLeakDetector"])

  name                = "${each.value} (tf deployed)"
  resource_group_name = var.application_insights.resource_group_name
  severity            = "Sev3"
  scope_resource_ids  = [var.application_insights.id]
  frequency           = "PT24H"
  detector_type       = each.value

  action_group {
    ids = [var.application_insights.action_group_id]
  }
}

```

### References

* [Migrate Azure Monitor Application Insights smart detection to alerts (Preview)](https://docs.microsoft.com/en-gb/azure/azure-monitor/alerts/alerts-smart-detections-migration)

Related
* #10563
* #8812

**Comment by teowa** [comment]
This issue has been fixed by PR #13998, if there is nothing else miss, maybe this issue can be closed @mbfrahry .

**Comment by mybayern1974** [comment]
@gjermystreeva , I'm closing this issue per teowa's comments above. Feel free to suggest to reopen it you found this feature is still unavailable in this provider.
